acq4/devices/DAQGeneric/DAQGeneric.py

```diff
@@ -39,9 +41,15 @@ def mapToDaq(self, chan, data):
         return (data*scale) - offset
         
     def mapFromDaq(self, chan, data):
-        scale = self.scale[chan]
-        offset = self.offset[chan]
-        return (data + offset) * scale
+        #print 'Mapping chan : ', self.device, chan, self.scale, self.offset, self.chantype
+        if 'ci' in self.chantype[chan]:
+            #print 'differentiated'
+            return np.diff(np.hstack((0,data)))
```

Comment:
OK. I moved the differentiation to 'devices/NiDAQ/nidaq.py: Task.getData()'. See last two commits in https://github.com/mgraupe/acq4/tree/photon-counting .